client/jetpack-connect/authorize.js

```diff
@@ -813,9 +805,7 @@ export class JetpackAuthorize extends Component {
 			return translate( 'Connect my site' );
 		}
 
-		if ( ! this.retryingAuth ) {
-			return translate( 'Approve' );
-		}
+		return translate( 'Approve' );
```

Comment:
The `this.retryingAuth` condition is handled a few lines above. If we get to this statement, it's already guaranteed to be `false`.